consensus/src/data_manager.rs

```diff
@@ -0,0 +1,225 @@
+// Copyright (c) Aptos
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::quorum_store::batch_reader::BatchReader;
+use crate::quorum_store::utils::RoundExpirations;
+use aptos_crypto::HashValue;
+use aptos_infallible::Mutex;
+use aptos_logger::{debug, warn};
+use aptos_types::transaction::SignedTransaction;
+use arc_swap::ArcSwapOption;
+use consensus_types::{
+    block::Block,
+    common::Payload,
+    proof_of_store::{LogicalTime, ProofOfStore},
+    request_response::WrapperCommand,
+};
+use dashmap::DashMap;
+use executor_types::*;
+use futures::channel::mpsc::Sender;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tokio::sync::oneshot;
+
+enum DataStatus {
+    Cached(Vec<SignedTransaction>),
+    Requested(Vec<oneshot::Receiver<Result<Vec<SignedTransaction>, Error>>>),
+}
+
+/// Execution -> QuorumStore notification of commits.
+pub struct DataManager {
+    quorum_store_enabled: AtomicBool,
+    data_reader: ArcSwapOption<BatchReader>,
+    quorum_store_wrapper_tx: ArcSwapOption<Sender<WrapperCommand>>,
+    digest_status: DashMap<HashValue, DataStatus>,
```

Comment:
Could the `digest_status` just be a part of `Payload::InQuorumStore`? What are we getting out of having a separate map?

Reply:
Done.